vars/packageBuildingPipelineDAOS.groovy

```diff
@@ -631,11 +631,16 @@ def call(Map pipeline_args) {
                                           git checkout ''' + pipeline_args.get('daos_test_branch',
                                                                                'origin/master') + '''
                                           git submodule update --init
-                                          make PR_REPOS="''' + env.JOB_NAME.split('/')[1] +
+                                          if ! make PR_REPOS="''' + env.JOB_NAME.split('/')[1] +
                                                '@' + env.BRANCH_NAME + ':' + env.BUILD_NUMBER +
                                                ' ' + pipeline_args.get('test_repos', '') + '''" \
                                                CHROOT_NAME=epel-7-x86_64                        \
-                                               -C utils/rpms chrootbuild'''
+                                               -C utils/rpms chrootbuild; then
+                                            grep 'No matching package to install'               \
+                                                 /var/lib/mock/epel-7-x86_64/result/root.log
+                                          else
+                                            exit ${PIPESTATUS[0]}
```

Comment:
Doesn't this factor out to:
```suggestion
                                            exit 0
```
since this code-path only gets here if the make is successful?  But then I don't see why the `else` branch is even needed:

```
$ (if ! true; then exit 2; fi); echo $?
0
$ (if ! false; then exit 2; fi); echo $?
2
```

But most importantly, I think a comment around this code would be super useful for somebody coming and looking at this for the first time or in a long time, and forgetting what the use-case is that is being covered here.  It could be very easy to imagine somebody not considering the bootstrap situation and deciding that this check could lead to false positives and want to remove it due to not understanding why it's here in the first place.  I could see myself very easily doing that.

Reply:
The else step only was needed for the inverted logic test.  I have removed it and added a comment as to why the grep is present.